src/main/java/com/epam/training/gen/ai/config/DeploymentConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.training.gen.ai.config;
+
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Configuration
+public class DeploymentConfig {
+
+    @Value("${client-azureopenai-valid-deployment-names}")
+    private String validDeploymentNames;
+
+    @Getter
+    private List<String> validDeploymentNamesList;
+
+    @PostConstruct
+    public void init() {
+        validDeploymentNamesList = Arrays.asList(validDeploymentNames.split(","));
+    }
+}
```

Comment:
I searched the Internet to find out how to set a list as a property, hoping to be useful. I suggest you look at this article: https://www.baeldung.com/spring-inject-arrays-lists

Reply:
Updated